holopy/inference/prior.py

```diff
@@ -412,6 +412,11 @@ def map_keys(self):
         return ((key, getattr(self, key)) for key in ['real', 'imag'])
 
 
+# Any new prior types should be added to:
+#   holopy.scattering.interface.PRIOR_TYPES
+#   holopy.scattering.tests.test_interface.TestValidateScatterer.priors
+
+
```

Comment:
I really like that you added this. Perhaps a better place for it though is in the `Prior` docstring, line 48, right below where it says
```python
"""
       Prior subclasses should define at least the following methods:
       
           - guess
           - sample
           - prob
           - lnprob
"""
```

Alternatively, this is probably one of the few good use cases for a metaclass in Python. You could make new classes of `Prior` automatically log themselves in `holopy.scattering.interface.PRIOR_TYPES`. However, there are so few prior classes here, and I don't know they are being created frequently enough to be worth the metaclass usage.

Reply:
I like the idea of a metaclass! This solves the issue of knowing what prior classes exist, but we also need an example instance of each prior class to test, and right now there's no way to generate that. We could have each prior class know about an example instance with reasonable parameter values, e.g. `Uniform(0, 1)`, `Gaussian(1, 0)`, but that would mean adding code to the prior.py file that is currently contained in `holopy.scattering.tests.test_interface.TestValidateScatterer.priors`, and I think it would be bit messy - it would also add an additional step to anyone defining a new prior class. Do you think that overhead is worth it?

Reply:
> we also need an example instance of each prior class to test, and right now there's no way to generate that.

I think that `TestValidateScatterer.test_all_priors_accounted_for` will solves this problem without the need for generating an example instance, as long as the `Prior` metaclass adds the class name to `holopy.scattering.interface.PRIOR_TYPES`. Then `TestValidateScatterer.test_all_priors_accounted_for` will raise an error if a prior type is in `holopy.scattering.interface.PRIOR_TYPES` and not in `TestValidateScatter.priors`. (Perhaps `TestValidateScatterer.priors` could be a dict or something to ensure that each class is represented, rather than just the lengths being the same. Likewise, perhaps the test gives a more descriptive error message to the developer if it fails.) I think adding to the tests is part of TDD, so the future developer should expect to do that anyways :)

Again, I think this may be the "right" way to do it, but I'm not sure if it's worth it given that there are only 4 prior types and I don't foresee a future proliferation of prior types in `holopy`. It's up to you.

Reply:
FYI, this video demonstrates what I was talking about regarding metaclasses (around the 20 minute mark and following)
https://www.youtube.com/watch?v=7lmCu8wz8ro
It's a bit long, but it's good if you're just killing time on a Friday night during the pandemic :smile:
(and I'll take a look at the changes you pushed soon)